api/source/service/utils.js

```diff
@@ -143,7 +143,7 @@ module.exports.initializeDatabase = async function () {
 module.exports.parseRevisionStr = function (revisionStr) {
   let ro = {}
   if (revisionStr !== 'latest') {
-    let results = /V(\d+)R(\d+(\.\d+)?)/.exec(revisionStr)
+    let results = extractRevisionDetails(revisionStr)
     ro.version = results[1]
```

Comment:
Since you are returning just version and release from your new function, we should be assigning results[0] and [1] here now instead of 1 and 2.